custom/enikshay/tests/test_adherence_updates.py

```diff
@@ -0,0 +1,363 @@
+                'expected_doses_taken': (2.0 / 7) * int(self.fixture_data['schedule1']),
+                'aggregated_score_count_taken': 2,
+                'adherence_latest_date_recorded': date(2016, 1, 12),
+                'adherence_total_doses_taken': 2
+            }
+        )
+
+    def test_unknown_adherence_data_less_and_greater(self):
+        self.assert_update(
+            (
+                datetime(2016, 1, 20),
+                (datetime(2016, 1, 10), 'schedule1'),
+                [
+                    (datetime(2016, 1, 11), DOSE_TAKEN_INDICATORS[0]),
+                    (datetime(2016, 1, 12), DOSE_TAKEN_INDICATORS[0]),
+                    (datetime(2016, 1, 14), 'unknown'),
+                    (datetime(2016, 1, 21), 'unknown')
+                ]
+            ),
+            {
+                'aggregated_score_date_calculated': date(2016, 1, 12),
+                'expected_doses_taken': (2.0 / 7) * int(self.fixture_data['schedule1']),
+                'aggregated_score_count_taken': 2,
+                'adherence_latest_date_recorded': date(2016, 1, 12),
+                'adherence_total_doses_taken': 2
+            }
+        )
+
+    def test_missed_adherence_dose(self):
+        self.assert_update(
+            (
+                datetime(2016, 1, 20),
+                (datetime(2016, 1, 10), 'schedule1'),
+                [
+                    (datetime(2016, 1, 11), DOSE_TAKEN_INDICATORS[0]),
+                    (datetime(2016, 1, 12), DOSE_TAKEN_INDICATORS[0]),
+                    (datetime(2016, 1, 14), 'unknown'),
+                    (datetime(2016, 1, 21), DOSE_TAKEN_INDICATORS[3])  # dose missed
+                ]
+            ),
+            {
+                'aggregated_score_date_calculated': date(2016, 1, 20),
+                'expected_doses_taken': (10.0 / 7) * int(self.fixture_data['schedule1']),
+                # ask clayton
+                'aggregated_score_count_taken': 2,
+                'adherence_latest_date_recorded': date(2016, 1, 21),
+                'adherence_total_doses_taken': 3
+            }
+        )
+
+    def test_two_doses_on_same_day(self):
+        self.assert_update(
+            (
+                datetime(2016, 1, 20),
+                (datetime(2016, 1, 10), 'schedule1'),
+                [
+                    # same day, different time
+                    (datetime(2016, 1, 11, 1), DOSE_TAKEN_INDICATORS[0]),
+                    (datetime(2016, 1, 11, 3), DOSE_TAKEN_INDICATORS[0]),
+                ]
+            ),
+            {
+                'aggregated_score_date_calculated': date(2016, 1, 11),
+                'expected_doses_taken': (1.0 / 7) * int(self.fixture_data['schedule1']),
+                'aggregated_score_count_taken': 1,
+                'adherence_latest_date_recorded': date(2016, 1, 11),
+                'adherence_total_doses_taken': 1
+            }
+        )
+
+    def test_two_doses_on_same_day_one_missed(self):
```

Comment:
One other potential test case could be two doses on the same day, but different indicators (IE: one directly observed, one self reported), on the same day, which should also be recorded as 1 dose (not two)